src/other/broadcasting.jl

```diff
@@ -0,0 +1,63 @@
+struct LazyNewColDataFrame
+    df::DataFrame
+    col::Symbol
+end
+
+# we allow LazyNewColDataFrame only for data frames with some columns
+Base.axes(x::LazyNewColDataFrame) = axes(x.df[1])
+
+Base.Broadcast.broadcastable(df::AbstractDataFrame) = Matrix(df)
+
+Base.maybeview(df::AbstractDataFrame, idxs...) = view(df, idxs...)
+
+function Base.maybeview(df::AbstractDataFrame, idxs)
+    if ncol(df) == 0
+        throw(ArgumentError("Broadcasting into a data frame with no columns is not allowed"))
+    end
+    if idxs isa Symbol
+        if !haskey(df, idxs)
+            if !(df isa DataFrame)
+                throw(ArgumentError("column $idxs not found"))
+            end
+            return LazyNewColDataFrame(df, idxs)
+        end
+    end
+    view(df, idxs)
+end
+
+function Base.copyto!(lazydf::LazyNewColDataFrame, bc)
+    if isempty(lazydf.df)
+        throw(ArgumentError("Broadcasting creation of a column of an empty data frame is not allowed"))
+    end
+    T = mapreduce(i -> typeof(bc[i]), promote_type, eachindex(bc); init=Union{})
+    col = Tables.allocatecolumn(T, nrow(lazydf.df))
+    copyto!(col, bc)
+    lazydf.df[lazydf.col] = col
+end
+
+function Base.copyto!(df::AbstractDataFrame, bc)
+    for I in eachindex(bc)
```

Comment:
I guess it would be more efficient to iterate column-wise. That way we wouldn't have to lookup the column for each entry, but only once.

Reply:
But we have no guarantee what `eachindex` returns and in what order, and `eachindex` is the only guaranteed iteration infrastructure for `bc`. In short: we cannot get a column from `bc` object.

Reply:
AFAICT according to https://docs.julialang.org/en/latest/manual/interfaces/#man-interfaces-broadcasting-1, `Broadcastable` objects have to implement `axes` and `getindex`, so isn't that all we need? It would be really too bad to support broadcasting but with a terrible performance...

Reply:
I understand you refer to:

> Iterating over the `CartesianIndices` of the `axes(::Broadcasted)` and using indexing with the resulting `CartesianIndex` object to compute the result.

I was afraid that there might be some strange `eachindex` methods for custom return values of `axes`. But it makes sense to assume that we can iterate over what `axes` returns. I will change this.